src/app/zap-templates/zcl/data-model/chip/energy-evse-cluster.xml

```diff
@@ -0,0 +1,235 @@
+    </command>
+
+
+    <event code="0x0000" name="EVConnected" priority="info" side="server" optional="false">
+      <description>Generated when the EV is plugged in.</description>
+      <field id="0" name="SessionId"  type="INT32U"         optional="false"/>
+    </event>
+
+    <event code="0x0001" name="EVNotDetected" priority="info" side="server" optional="false">
+      <description>Generated when the EV is unplugged or not detected (having been previously plugged in). When the vehicle is unplugged then the session is ended.</description>
+      <field id="0" name="SessionId"               type="INT32U"         optional="false"/>
+      <field id="1" name="State"                   type="EvseStateEnum"  optional="false"/>
+      <field id="2" name="SessionDuration"         type="elapsed_s"      optional="false"/>
+      <field id="3" name="SessionEnergyCharged"    type="INT32S"         optional="false"/>
+      <field id="4" name="SessionEnergyDischarged" type="INT32S"         optional="false"/>
+    </event>
+
+    <event code="0x0002" name="EnergyTransferStarted" priority="info" side="server" optional="false">
+      <description>Generated when the EV starts charging or discharging (V2X only).</description>
+      <field id="0" name="SessionId"               type="INT32U"         optional="false"/>
+      <field id="1" name="State"                   type="EvseStateEnum"  optional="false"/>
+      <field id="2" name="MaximumCurrent"          type="INT32U"         optional="false"/>
+    </event>
+
+    <event code="0x0003" name="EnergyTransferStopped" priority="info" side="server" optional="false">
+      <description>Generated when the EV stops charging or discharging (V2X only).</description>
+      <field id="0" name="SessionId"         type="INT32U"         optional="false"/>
+      <field id="1" name="State"             type="EvseStateEnum"  optional="false"/>
+      <field id="2" name="Reason"            type="EnergyTransferStoppedReasonEnum" optional="false"/>
+      <field id="4" name="EnergyTransferred" type="INT32S"         optional="false"/>
+    </event>
+
+    <event code="0x0004" name="Fault" priority="info" side="server" optional="false">
+      <description>If the EVSE detects a fault, then it SHALL generate a Fault Event.</description>
+      <field id="0" name="SessionId"                type="INT32U"         optional="false"/>
+      <field id="1" name="State"                    type="EvseStateEnum"  optional="false"/>
+      <field id="2" name="FaultStatePreviousState"  type="EvseStateEnum"  optional="false"/>
+      <field id="4" name="FaultStateCurrentState"   type="EvseStateEnum"  optional="false"/>
+    </event>
+
+    <event code="0x0005" name="RFID" priority="info" side="server" optional="true">
+      <description>Generated when a RFID card has been read.</description>
+      <field id="0" name="UID"  type="OCTET_STRING" length="10" optional="false" isNullable="true"/>
```

Comment:
Spec seems to both say "not nullable" and "default is null" here ... which seems to be a spec mistake. Could you update the spec to add a `X` quality?

Reply:
Hmm - don't think it should have been nullable - so will update the spec to remove the default and will change this to isNullable="false".